File: leetcode/backspace-string-compare.py
class Solution:
    def backspaceCompare(self, s: str, t: str) -> bool:
        sr = ""
        ar = ""
        fs = ""
        fa = ""
        for x in range(len(s)-1, -1, -1):
            sr +=s[x]
        for x in range(len(t)-1,-1,-1):
            ar += t[x]
        skip = 0
        for char in sr:
            if char == "#":
                skip += 1
            else:
                if skip == 0:
                    fs += char
                if skip > 0:
                    skip -= 1
        skip = 0
        for char in ar:
            if char == "#":
                skip += 1
            else:
                if skip == 0:
                    fa += char
                if skip > 0:
                    skip -= 1
        if fa == fs:
            return True
        return False

File: leetcode/backspace-string-compare.cpp
#include <bits/stdc++.h>
#include <algorithm>
using namespace std;

class Solution {
public:
    bool backspaceCompare(string s, string t) {
        string rs;
        string rt;
        string fs;
        string ft;
        
        for(int x = s.size()-1;x!=-1;x--){
            rs += s[x];
        }
        for(int x = t.size()-1;x!=-1;x--){
            rt += t[x];
        }
        int skip = 0;
        for(char c:rs){
            if(c=='#'){
                skip++;
            }
            else{
                if(skip==0){
                    fs += c;
                }
                else if(skip>0){
                    skip--;
                }
            }
        }
        skip =0;
        for(char c:rt){
            if(c=='#'){
                skip++;
            }
            else{
                if(skip==0){
                    ft += c;
                }
                else if(skip>0){
                    skip--;
                }
            }
        }
        return ft==fs;
        
    }
};